src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -358,6 +358,10 @@ private BuildData MergeBuildManifests(List<BuildData> buildsMetadata)
                 mergedBuild.Assets = mergedBuild.Assets.AddRange(build.Assets);
             }
 
+            // Remove duplicated assets based on the top level properties of the asset.
+            // The AssetLocations property isn't set at this point yet.
+            mergedBuild.Assets = mergedBuild.Assets.Distinct().ToImmutableList();
```

Comment:
I don't think this works, because I think the default object comparer that will be used will say that the two assets are actually different. I'd write a test for this.

Reply:
Added a AssetDataComparer.